### PR TITLE
Add CLI

### PR DESCRIPTION
**Pull request by gabrielmbmb**
## Description

This PR adds a CLI application to `distilabel` that allows (for now) to:

1. Run a pipeline given the config YAML file that can be either a local file or a URL:
    ```bash
    distilabel pipeline run --config pipeline.yaml \
	    --param load_hub_dataset.repo_id=HuggingFaceH4/instruction-dataset \
	    --param load_hub_dataset.split=test \
	    --param text_generation_gpt.num_generations=3 \
	    --param text_generation_gpt_2.num_generations=3 \
	    --param push_to_hub.repo_id=distilabel-internal-testing/testing-distilabel-push-to-hub \
	    --param push_to_hub_2.repo_id=distilabel-internal-testing/testing-distilabel-push-to-hub-2 \
	    --ignore-cache \
	    --repo-id distilabel-internal-testing/testing-distilabel-cli \
	    --commit-message "Testing pipeline"
    ```
2. Get the information of a pipeline (name, description, steps, runtime parameters, etc) given the config YAML file or a URL:
    ```bash
    distilabel pipeline info --config pipeline.yaml
    ```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -24,18 +24,22 @@ dependencies = [
     "datasets >= 2.14.0",
     "importlib-resources >= 6.1.1; python_version < '3.9'",
     "Jinja2 >= 3.1.2",
+    "pydantic >= 2.0",
+    "networkx >= 3.0",
+    "scipy >= 1.10.0",
```
Comment (alvarobartt) [inline pyproject.toml:29]: Duplicated dependency
```suggestion
```

File: pyproject.toml
Comment (alvarobartt) [inline pyproject.toml:1]: Can we keep the `dependencies` alphabetical sorting?

Comment (gabrielmbmb) [reply pyproject.toml:1]: 🤭 yes

File: src/distilabel/pipeline/base.py [hunk 2]
```diff
@@ -172,14 +196,30 @@ def run(self, parameters: Optional[Dict[str, Dict[str, Any]]] = None) -> "Distis
 
         Args:
             parameters: A dictionary with the step name as the key and a dictionary with
-                the parameter name as the key and the parameter value as the value.
+                the runtime parameters for the step as the value. Defaults to `None`.
+            use_cache: whether to use the cache from previous pipeline runs. Defaults to
```
Comment (plaguss) [inline src/distilabel/pipeline/base.py:200]:
```suggestion
            use_cache: Whether to use the cache from previous pipeline runs. Defaults to
```

File: src/distilabel/pipeline/local.py [hunk 3]
```diff
@@ -48,22 +48,26 @@
...
         """Runs the pipeline.
 
         Args:
-            parameters: a dictionary containing the runtime parameters for each step.
-                The keys are the step names and the values are dictionaries in which the
-                keys are the parameter names (defined in the `process` method of the step)
-                and the values are the parameter values.
+            parameters: A dictionary with the step name as the key and a dictionary with
+                the runtime parameters for the step as the value. Defaults to `None`.
+            use_cache: whether to use the cache from previous pipeline runs. Defaults to
```
Comment (plaguss) [inline src/distilabel/pipeline/local.py:61]:
```suggestion
            use_cache: Whether to use the cache from previous pipeline runs. Defaults to
```

**Pull request merged**